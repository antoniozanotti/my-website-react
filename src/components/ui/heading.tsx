import { TzHeading } from "topaz-react";

export interface HeadingProps {
  level: "h1" | "h2" | "h3" | "h4";
  title: string;
  anchor?: string;
  className?: string;
}

export default function Heading({
  level,
  title,
  anchor = "",
  className = "",
}: HeadingProps) {
  let size: any = "";
  let variant: any = "";
  let headingClasses = className + " drop-shadow";
  switch (level) {
    case "h1":
      size = "large";
      variant = "accent";
      break;
    case "h2":
      size = "large";
      variant = "accent-1";
      headingClasses += " mt-[164px]";
      break;
    case "h3":
      size = "medium";
      variant = "accent-2";
      headingClasses += " mt-[68px]";
      break;
    default:
      size = "small";
      variant = "secondary";
      headingClasses += " mt-[56px]";
      break;
  }
  return (
    <TzHeading
      level={level}
      size={size}
      variant={variant}
      className={headingClasses}
      id={anchor}
    >
      {title}
    </TzHeading>
  );
}
