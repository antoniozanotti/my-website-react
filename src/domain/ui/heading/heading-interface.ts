import { LevelEnum } from "./enums/level-enum";

export interface HeadingInterface {
  title: string;
  level?: keyof typeof LevelEnum;
  anchor?: string;
  className?: string;
}