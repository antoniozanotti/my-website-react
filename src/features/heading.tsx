import React from "react";
import { ReactNode } from "react";
import { TzHeading, TzHeadingProps } from "topaz-react";

export interface HeadingProps {
  level: "h1" | "h2" | "h3" | "h4";
  title: string;
}

export const Heading = ({ level, title }: HeadingProps) => {
  let size: any = "large";
  let variant: any = "accent";
  switch (level) {
    case "h2":
      variant = "accent-1";
      break;
    case "h3":
      size = "medium";
      variant = "accent-2";
      break;
    case "h4":
      size = "small";
      variant = "primary-2";
      break;
  }
  return (
    <TzHeading level={level} size={size} variant={variant} className="drop-shadow-lg">
      {title}
    </TzHeading>
  );
};
