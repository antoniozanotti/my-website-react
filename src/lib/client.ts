import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
