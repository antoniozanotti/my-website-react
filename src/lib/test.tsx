import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
