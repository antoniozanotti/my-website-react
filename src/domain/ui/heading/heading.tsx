import { TzHeading } from "topaz-react";
import { HeadingInterface } from "./heading-interface";
import { LevelEnum } from "./enums/level-enum";
import type { CustomStylesType } from "./types/custom-styles-type";
import { SizeEnum } from "./enums/size-enum";
import { VariantEnum } from "./enums/variant-enum";

export function Heading({
  title,
  level = "h1",
  anchor = "",
  className = "",
}: HeadingInterface) {
  const baseStyles = `drop-shadow`;
  const { size, variant, marginStyles } = useCustomStyles(level);
  const headingStyles = `${baseStyles} ${marginStyles} ${className}`;
  return (
    <TzHeading
      level={level}
      size={size}
      variant={variant}
      className={headingStyles}
      id={anchor}
    >
      {title}
    </TzHeading>
  );
}

function useCustomStyles(level: keyof typeof LevelEnum): CustomStylesType {
  return {
    h1: {
      size: SizeEnum.large,
      variant: VariantEnum["accent"],
      marginStyles: "",
    },
    h2: {
      size: SizeEnum.large,
      variant: VariantEnum["accent-1"],
      marginStyles: "mt-[164px]",
    },
    h3: {
      size: SizeEnum.medium,
      variant: VariantEnum["accent-2"],
      marginStyles: "mt-[68px]",
    },
    h4: {
      size: SizeEnum.small,
      variant: VariantEnum["secondary"],
      marginStyles: "mt-[56px]",
    },
  }[level];
}
