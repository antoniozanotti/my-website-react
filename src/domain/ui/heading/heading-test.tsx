import { screen, render, firstComponent } from "@/lib/test";
import { Heading } from "./heading";

describe("Heading", () => {
  test("should render default props", async () => {
    render(<Heading title="Test" />);
    const component = firstComponent(screen);

    expect(component).toHaveTextContent("Test");
  });

  test("should render h1", async () => {
    render(<Heading title="Test" level="h1" />);
    const component = screen.getByRole("heading", {level: 1});

    expect(component).toHaveTextContent("Test");
  });

  test("should render h2", async () => {
    render(<Heading title="Test" level="h2" />);
    const component = screen.getByRole("heading", {level: 2});

    expect(component).toHaveTextContent("Test");
  });

  test("should render h3", async () => {
    render(<Heading title="Test" level="h3" />);
    const component = screen.getByRole("heading", {level: 3});

    expect(component).toHaveTextContent("Test");
  });

  test("should render h4", async () => {
    render(<Heading title="Test" level="h4" />);
    const component = screen.getByRole("heading", {level: 4});

    expect(component).toHaveTextContent("Test");
  });

  test("should accept anchor", async () => {
    render(<Heading title="Test" anchor="testid" />);
    const component = firstComponent(screen);

    expect(component).toHaveAttribute("id", "testid");
  });

  test("should accept className", async () => {
    render(<Heading title="Test" className="test" />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/test$/);
  });
});
