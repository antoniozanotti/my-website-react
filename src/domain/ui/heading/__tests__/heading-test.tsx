import { screen, render, firstComponent } from "@/lib/test";
import { Heading } from "../heading";

describe("Heading", () => {
  it("should render default props", async () => {
    const { container } = render(<Heading title="Test" />);
    const component = firstComponent(screen);

    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });

  it("should render h1", async () => {
    const { container } = render(<Heading title="Test" level="h1" />);
    const component = screen.getByRole("heading", {level: 1});

    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });

  it("should render h2", async () => {
    const { container } = render(<Heading title="Test" level="h2" />);
    const component = screen.getByRole("heading", {level: 2});

    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });

  it("should render h3", async () => {
    const { container } = render(<Heading title="Test" level="h3" />);
    const component = screen.getByRole("heading", {level: 3});

    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });

  it("should render h4", async () => {
    const { container } = render(<Heading title="Test" level="h4" />);
    const component = screen.getByRole("heading", {level: 4});

    expect(component).toHaveTextContent("Test");
    expect(container).toMatchSnapshot();
  });

  it("should accept anchor", async () => {
    const { container } = render(<Heading title="Test" anchor="testid" />);
    const component = firstComponent(screen);

    expect(component).toHaveAttribute("id", "testid");
    expect(container).toMatchSnapshot();
  });

  it("should accept className", async () => {
    const { container } = render(<Heading title="Test" className="test" />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/test$/);
    expect(container).toMatchSnapshot();
  });
});
