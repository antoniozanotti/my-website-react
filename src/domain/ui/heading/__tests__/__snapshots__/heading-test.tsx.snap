// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should accept anchor 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h1
          class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent dark:text-dark-accent drop-shadow  "
          id="testid"
        >
          Test
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Heading should accept className 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h1
          class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent dark:text-dark-accent drop-shadow  test"
          id=""
        >
          Test
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Heading should render default props 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h1
          class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent dark:text-dark-accent drop-shadow  "
          id=""
        >
          Test
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Heading should render h1 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h1
          class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent dark:text-dark-accent drop-shadow  "
          id=""
        >
          Test
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Heading should render h2 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h2
          class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
          id=""
        >
          Test
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`Heading should render h3 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h3
          class="leading-none text-c7 sm:text-c8 lg:text-c10 text-accent-2 dark:text-dark-accent-2 drop-shadow mt-[68px] "
          id=""
        >
          Test
        </h3>
      </div>
    </div>
  </div>
</div>
`;

exports[`Heading should render h4 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <h4
          class="leading-none text-c5 md:text-c6 lg:text-c7 xl:text-c8 text-secondary dark:text-dark-secondary drop-shadow mt-[56px] "
          id=""
        >
          Test
        </h4>
      </div>
    </div>
  </div>
</div>
`;
